pkg/daemon/update.go

```diff
@@ -2026,3 +2029,163 @@ func (dn *Daemon) reboot(rationale string) error {
 	MCDRebootErr.WithLabelValues(dn.node.Name, "reboot failed", "this error should be unreachable, something is seriously wrong").SetToCurrentTime()
 	return fmt.Errorf("reboot failed; this error should be unreachable, something is seriously wrong")
 }
+
+// experimentalUpdateLayeredConfig() pretends to do the normal config update for the pool but actually does
+// an image update instead. This function should be completely thrown away.
+// TODO(jkyros): right now this skips drain and reboot, it just live-applies it, but you *can* boot it and it will work
+func (dn *Daemon) experimentalUpdateLayeredConfig() error {
+
+	desiredImage := dn.node.Annotations[constants.DesiredImageConfigAnnotationKey]
+	currentImage := dn.node.Annotations[constants.CurrentImageConfigAnnotationKey]
+	desiredConfig := dn.node.Annotations[constants.DesiredMachineConfigAnnotationKey]
+
+	// Layered doesn't exist right out of the gate right now, it takes some time to reconcile
+	// We should never get here now though, since node-controller is waiting for the image
+	if desiredImage == "" {
+		glog.Infof("Looks like we don't have a desired image yet. Nothing to do.")
+		return nil
+	}
+
+	// I don't set currentImage ever, so we always hit the else block right now
+	if currentImage == desiredImage {
+		// Orrrr....if we've live updated to it
+		glog.Infof("Node is on proper image %s", desiredImage)
+
+		glog.Infof("Completing pending config %s", currentImage)
+		if err := dn.completeUpdate(currentImage); err != nil {
+			MCDUpdateState.WithLabelValues("", err.Error()).SetToCurrentTime()
+
+		}
+
+		// TODO(jkyros): For now I'm just making the pool happy so it's like "yeah I'm done"
+		if err := dn.nodeWriter.SetDone(dn.kubeClient.CoreV1().Nodes(), dn.nodeLister, dn.name, desiredConfig); err != nil {
+			errLabelStr := fmt.Sprintf("error setting node's state to Done: %v", err)
+			MCDUpdateState.WithLabelValues("", errLabelStr).SetToCurrentTime()
+			return nil
+		}
+
+	} else {
+		// We think we have work to do
+
+		client := &RpmOstreeClient{}
+		state, err := client.GetState()
+		if err != nil {
+			return err
+		}
+
+		// Look through our deployments
+		// If we've rebased to it, but not booted, that's okay IF we've live-applied
+		for _, deployment := range state.Deployments {
+
+			// What we're looking for is at least in the list
+			if strings.TrimPrefix(deployment.ContainerImageReference, "ostree-unverified-registry:") == desiredImage {
```

Comment:
I think we should parse this more properly by splitting on the first `:`.  For now we can probably just error out if it doesn't start with `ostree-unverified-registry`; but down the line we should probably support the others.